File: backend/app/schemas/ultimate_platform.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


UltimateStatus = Literal["ready", "partial", "missing", "failed"]
UltimateVerdict = Literal[
    "COMPLETE AUTONOMOUS ENTERPRISE INTELLIGENCE & SIMULATION PLATFORM",
    "ULTIMATE PLATFORM GAPS REMAIN",
]


class PlatformAuditMap(BaseModel):
    backend_files: int = Field(ge=0)
    frontend_files: int = Field(ge=0)
    api_route_modules: int = Field(ge=0)
    service_modules: int = Field(ge=0)
    schema_modules: int = Field(ge=0)
    ai_modules: int = Field(ge=0)
    dashboard_components: int = Field(ge=0)
    persisted_data_stores: int = Field(ge=0)
    dependency_files: list[str]
    api_map: list[str]
    database_map: list[str]
    frontend_component_map: list[str]
    ai_module_map: list[str]


class UltimateFeatureAudit(BaseModel):
    feature_id: int = Field(ge=1, le=15)
    name: str
    status: UltimateStatus
    present: bool
    working: bool
    connected: bool
    tested: bool
    production_ready: bool
    score: float = Field(ge=0, le=100)
    evidence: list[str]
    integrations: list[str]
    endpoints: list[str]
    dashboards: list[str]


class IntegrationAuditLink(BaseModel):
    source: str
    target: str
    status: UltimateStatus
    evidence: list[str]


class VirtualEmployeeProfile(BaseModel):
    employee_id: str
    name: str
    role: str
    department: str
    behavior_model: str
    work_pattern: str
    productivity_profile: float = Field(ge=0, le=100)
    collaboration_profile: float = Field(ge=0, le=100)
    stress_propagation_risk: float = Field(ge=0, le=100)
    leadership_effect: float = Field(ge=0, le=100)


class TimeMachineScenario(BaseModel):
    question: str
    horizon_months: int = Field(ge=1, le=36)
    burnout_forecast: float = Field(ge=0, le=100)
    revenue_impact_percent: float
    productivity_impact_percent: float
    attrition_risk: float = Field(ge=0, le=100)
    project_delay_probability: float = Field(ge=0, le=100)
    team_health_score: float = Field(ge=0, le=100)
    recommendation: str


class GlobalRiskSignal(BaseModel):
    risk: str
    category: str
    severity: Literal["low", "medium", "high", "critical"]
    score: float = Field(ge=0, le=100)
    strategic_insight: str
    recommended_action: str
    source_systems: list[str]


class ProductionReadinessReport(BaseModel):
    score: float = Field(ge=0, le=100)
    authentication: UltimateStatus
    authorization: UltimateStatus
    logging: UltimateStatus
    monitoring: UltimateStatus
    error_handling: UltimateStatus
    ci_cd: UltimateStatus
    evidence: list[str]


class UltimatePlatformScorecard(BaseModel):
    judge_wow_factor_score: float = Field(ge=0, le=100)
    innovation_score: float = Field(ge=0, le=100)
    enterprise_score: float = Field(ge=0, le=100)
    integration_score: float = Field(ge=0, le=100)
    security_score: float = Field(ge=0, le=100)
    performance_score: float = Field(ge=0, le=100)
    production_readiness_score: float = Field(ge=0, le=100)
    minimum_score: float = Field(ge=0, le=100)


class UltimatePlatformResponse(BaseModel):
    model: str
    generated_at: datetime
    audit_map: PlatformAuditMap
    feature_coverage_report: list[UltimateFeatureAudit]
    integration_report: list[IntegrationAuditLink]
    error_report: list[str]
    security_report: list[str]
    performance_report: list[str]
    production_readiness_report: ProductionReadinessReport
    virtual_employees: list[VirtualEmployeeProfile]
    time_machine_scenarios: list[TimeMachineScenario]
    global_risk_signals: list[GlobalRiskSignal]
    scorecard: UltimatePlatformScorecard
    missing_components: list[str]
    fixed_components: list[str]
    regenerated_components: list[str]
    final_verdict: UltimateVerdict
    source_systems: list[str]
    storage: str
    stream_sequence: int = 1

File: frontend/src/types/ultimate-platform.ts
export type UltimateStatus = "ready" | "partial" | "missing" | "failed";
export type UltimateVerdict =
  | "COMPLETE AUTONOMOUS ENTERPRISE INTELLIGENCE & SIMULATION PLATFORM"
  | "ULTIMATE PLATFORM GAPS REMAIN";

export interface PlatformAuditMap {
  backendFiles: number;
  frontendFiles: number;
  apiRouteModules: number;
  serviceModules: number;
  schemaModules: number;
  aiModules: number;
  dashboardComponents: number;
  persistedDataStores: number;
  dependencyFiles: string[];
  apiMap: string[];
  databaseMap: string[];
  frontendComponentMap: string[];
  aiModuleMap: string[];
}

export interface UltimateFeatureAudit {
  featureId: number;
  name: string;
  status: UltimateStatus;
  present: boolean;
  working: boolean;
  connected: boolean;
  tested: boolean;
  productionReady: boolean;
  score: number;
  evidence: string[];
  integrations: string[];
  endpoints: string[];
  dashboards: string[];
}

export interface IntegrationAuditLink {
  source: string;
  target: string;
  status: UltimateStatus;
  evidence: string[];
}

export interface VirtualEmployeeProfile {
  employeeId: string;
  name: string;
  role: string;
  department: string;
  behaviorModel: string;
  workPattern: string;
  productivityProfile: number;
  collaborationProfile: number;
  stressPropagationRisk: number;
  leadershipEffect: number;
}

export interface TimeMachineScenario {
  question: string;
  horizonMonths: number;
  burnoutForecast: number;
  revenueImpactPercent: number;
  productivityImpactPercent: number;
  attritionRisk: number;
  projectDelayProbability: number;
  teamHealthScore: number;
  recommendation: string;
}

export interface GlobalRiskSignal {
  risk: string;
  category: string;
  severity: "low" | "medium" | "high" | "critical";
  score: number;
  strategicInsight: string;
  recommendedAction: string;
  sourceSystems: string[];
}

export interface ProductionReadinessReport {
  score: number;
  authentication: UltimateStatus;
  authorization: UltimateStatus;
  logging: UltimateStatus;
  monitoring: UltimateStatus;
  errorHandling: UltimateStatus;
  ciCd: UltimateStatus;
  evidence: string[];
}

export interface UltimatePlatformScorecard {
  judgeWowFactorScore: number;
  innovationScore: number;
  enterpriseScore: number;
  integrationScore: number;
  securityScore: number;
  performanceScore: number;
  productionReadinessScore: number;
  minimumScore: number;
}

export interface UltimatePlatformResponse {
  model: string;
  generatedAt: string;
  auditMap: PlatformAuditMap;
  featureCoverageReport: UltimateFeatureAudit[];
  integrationReport: IntegrationAuditLink[];
  errorReport: string[];
  securityReport: string[];
  performanceReport: string[];
  productionReadinessReport: ProductionReadinessReport;
  virtualEmployees: VirtualEmployeeProfile[];
  timeMachineScenarios: TimeMachineScenario[];
  globalRiskSignals: GlobalRiskSignal[];
  scorecard: UltimatePlatformScorecard;
  missingComponents: string[];
  fixedComponents: string[];
  regeneratedComponents: string[];
  finalVerdict: UltimateVerdict;
  sourceSystems: string[];
  storage: string;
  streamSequence: number;
}
